acceptance/builder_test.go

```diff
@@ -0,0 +1,412 @@
+					copyDir,
+					ctrPath("/layers"),
+					builderImage,
+					h.WithFlags(
+						"--env", "CNB_PLATFORM_API="+latestPlatformAPI,
+						"--env", "CNB_LAYERS_DIR=/layers/different_layer_dir_from_env",
+						"--env", "CNB_PLAN_PATH=/cnb/plan_tomls/always_detect_plan_buildpack_2.toml",
+					),
+				)
+				md := getBuilderMetadata(t, filepath.Join(copyDir, "layers/different_layer_dir_from_env/config/metadata.toml"))
+
+				h.AssertStringContains(t, md.Buildpacks[0].API, "0.2")
+				h.AssertStringContains(t, md.Buildpacks[0].ID, "hello_world_2")
+				h.AssertStringContains(t, md.Buildpacks[0].Version, "0.0.2")
+			})
+		})
+
+		// .../cmd/lifecycle/builder.go#Args
+		when("plan.toml path is not specified", func() {
+			it("will look for plan.toml in the provided layers directory", func() {
+				h.DockerRunAndCopy(t,
+					containerName,
+					copyDir,
+					ctrPath("/layers"),
+					builderImage,
+					h.WithFlags(
+						"--env", "CNB_PLATFORM_API="+latestPlatformAPI,
+						"--env", "CNB_LAYERS_DIR=/layers/different_layer_dir_from_env",
+						"--env", "CNB_GROUP_PATH=/cnb/group_tomls/always_detect_group_buildpack2.toml",
+					),
+				)
+				md := getBuilderMetadata(t, filepath.Join(copyDir, "layers/different_layer_dir_from_env/config/metadata.toml"))
+
+				h.AssertStringContains(t, md.Buildpacks[0].API, "0.2")
+				h.AssertStringContains(t, md.Buildpacks[0].ID, "hello_world_2")
+				h.AssertStringContains(t, md.Buildpacks[0].Version, "0.0.2")
+			})
+		})
+	})
+
+	when("CNB_APP_DIR changed", func() {
+		it("sets the buildpacks' working directory to CNB_APP_DIR", func() {
+			command := exec.Command(
+				"docker",
+				"run",
+				"--rm",
+				"--env", "CNB_PLATFORM_API="+latestPlatformAPI,
+				"--env", "CNB_GROUP_PATH=/cnb/group_tomls/always_detect_group.toml",
+				"--env", "CNB_PLAN_PATH=/cnb/plan_tomls/always_detect_plan.toml",
+				"--env", "CNB_APP_DIR=/env_folders/different_cnb_app_dir_from_env",
+				builderImage,
+			)
+			output, err := command.CombinedOutput()
+			h.AssertNil(t, err)
+			expected := "CNB_APP_DIR: /env_folders/different_cnb_app_dir_from_env"
+			h.AssertStringContains(t, string(output), expected)
+		})
+	})
+
+	when("CNB_BUILDPACKS_DIR changed", func() {
+		it("uses buildpacks from CNB_BUILDPACKS_DIR", func() {
+			command := exec.Command(
+				"docker",
+				"run",
+				"--rm",
+				"--env", "CNB_PLATFORM_API="+latestPlatformAPI,
+				"--env", "CNB_GROUP_PATH=/cnb/group_tomls/always_detect_group.toml",
+				"--env", "CNB_PLAN_PATH=/cnb/plan_tomls/always_detect_plan.toml",
+				"--env", "CNB_BUILDPACKS_DIR=/env_folders/different_buildpack_dir_from_env",
+				builderImage,
+			)
+			output, err := command.CombinedOutput()
+			h.AssertNil(t, err)
+			expected := "CNB_BUILDPACK_DIR: /env_folders/different_buildpack_dir_from_env"
+			h.AssertStringContains(t, string(output), expected)
+		})
+	})
+
+	when("CNB_LAYERS_DIR is set", func() {
+		it("CNB_LAYERS_DIR is successfully transmitted to build script", func() {
```

Comment:
```suggestion
		it("CNB_LAYERS_DIR is a parent of the buildpack layers dir", func() {
```
(this is more accurate, as it's not set in the env)